src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -329,3 +329,20 @@ internal inline fun <R> runOutsideIgnoredSection(descriptor: ThreadDescriptor?,
 
 internal const val LINCHECK_PACKAGE_NAME = "org.jetbrains.kotlinx.lincheck."
 internal const val LINCHECK_RUNNER_PACKAGE_NAME = "org.jetbrains.kotlinx.lincheck.runner."
+
+internal fun <T> Class<T>.newInstanceRecursive(): T {
```

Comment:
The name `newInstanceRecursive` is a bit misleading IMO. 
We just want to create some "default" instance of the class. We do not really care if it is inner or not, it is mostly an implementation detail.  

So maybe rename to something like `newDefaultInstance()` ?